grader/src/main/java/edu/pdx/cs410J/grader/scoring/ProjectSubmissionOutputFileParser.java

```diff
@@ -1,27 +1,106 @@
 package edu.pdx.cs410J.grader.scoring;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.Reader;
+import java.io.StringReader;
+import java.nio.CharBuffer;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
 import java.util.Date;
+import java.util.Locale;
+import java.util.regex.Pattern;
 
 /**
  * Author: Susham Kumar
  */
 
 public class ProjectSubmissionOutputFileParser {
-  public ProjectSubmissionOutputFileParser(Reader source) {
 
+  public ProjectSubmissionOutputFileParser(Reader source) {
+      if(source != null) {
+          bufferedReader = new BufferedReader(source);
+      }
 
   }
 
-  public ProjectSubmission parse() {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ProjectSubmission parse()  {
+
+     projectSubmission = new ProjectSubmission();
+     String fileContents=null;
+
+      try {
```

Comment:
You want to be sure to `close` the `BufferedReader` when your done with it.  I recommend using the "enhanced try/catch" syntax described here:

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html